### ISSUE TITLE
Sustainability of the project? Paywall? Is it actually free?

### ISSUE THREAD

**Issue by LokyStaz** [AlfishSoftware/godot-files-vscode#3]
I noticed in the "Support" section of your readme that you describe the software as "free." However, a significant portion of its functionality is gated behind a donation system.
While I understand and respect the need for financial support, this approach feels at odds with the philosophy of the Godot community, which emphasizes free access to tools, resources, and even assets to empower all developers equally.

For example, highly useful plugins like Godot Tools are distributed freely, allowing everyone to benefit regardless of their financial situation. By contrast, locking key features behind a paywall creates a barrier that feels inconsistent with the open and accessible nature of both Godot and its ecosystem.

Additionally, the fact that your project has no visible contributions and seems slated for abandonment in February 2025 raises concerns. It’s hard to justify financial support for a plugin that may no longer be maintained in a few months.

Crowdfunding isn't about threatening to abandon the project!
I appreciate the effort and skill that went into creating this, but I hope you'll consider aligning your project more closely with the spirit of the Godot community by making it fully accessible to all.
**That would encourage contributions (in term of code and donation), and ensure the longevity of the project.**

**Comment by AlfishSoftware** [comment]
I kindly ask that you please take the time to read everything for context.

### Is this project free?

> I noticed in the "Support" section of your readme that you describe the software as "free." However, a significant portion of its functionality is gated behind a donation system.

Yes, the project is indeed free, and in fact MORE free than other MIT-licensed projects such as Godot and godot-tools, because it's under The Unlicense (public domain). That's for the entire repo, ***including the features restricted to early access***. All of the code and build recipe is public. This means anyone can use all of the code today, and could even easily build their own copy of the extension with the restrictions removed (it's literally just a bunch of easy to find if-return statements), and even publish it without giving me any attribution! You can't get more in spirit of free software than that.
By "can" I mean it's both legally allowed and easy to do so.

See also:
- https://www.fsf.org/blogs/community/free-software-is-not-antithetical-to-commercial-success
- https://www.reddit.com/r/opensource/comments/k7kj4m/can_something_paid_be_called_open_source/
- https://opensource.stackexchange.com/questions/9937/is-my-software-still-free-if-i-charge-for-its-commercial-use-and-not-for-non-c

### What do you mean by "abandoning" the extension?

- The existing code will remain publicly accessible and in public domain forever. I don't intend to unpublish it.
- Bug fixes would likely still happen if needed.
- No time will be wasted to develop new features, except maybe if I were to put them on an actual paywall -- meaning those would be in a separate optional companion extension, paid and proprietary.
- If anyone wants to fork the project (free or commercially) or contribute to this, they will still be legally allowed to.

### Why "restricting" features?

> locking key features behind a paywall

The original intention of the "early access" system was to be an incentive to develop new features. The idea was that new features initially take the place of previous restricted features, making the oldest ones unrestricted. So the restricted part would be always shifted, and all features would eventually reach the public automatically for free, even those who never contributed.

This system is similar to e.g. youtubers with a Patreon early access (except I ask for a one-time donation, not a subscription). The idea was that some people would want early access and their contribution would benefit me, them, and everyone (by making development possible) at the same time.

An additional benefit of this system when used for software is that new features (thus less-tested, and can still have undetected bugs) won't affect the majority of users until later, when those bugs are found and fixed before most people even encounter them.

If donations were enough that I could focus work on this extension, it would work great. More payers mean more and faster features for non-payers too. But the idea turned out to be a complete failure, see below.

### Why are you asking for money so insistently?

> the fact that your project has no visible contributions and seems slated for abandonment in February 2025

Yes, I'm the sole developer of the project, you can see in the commits that I've received no help at all.

The deadline was January 31st, I've just recently pushed it another month because I'm really trying to not abandon it (I need to focus on my game because I have to publish it by January). Trust me, I completely understand you, and didn't want to resort to this at all. But I truly don't have any choice. Please let me give you a bit of context.

I'm trying to be an indie game developer, working alone (no social skills to get other people to help me; turns out that is way more important than getting good grades at school). I still don't have any published games yet, just 2 Unity assets and a Unity plugin that give me very occasional revenue (about $300~$350/year). I don't have any other job, and haven't had one in years, as I wasted time on abandoned projects as I was trying to learn Unity and Godot.

Let me share some stats I had calculated before starting the crowdfunding campaign (I thought about putting them in the README, but ended up not doing it, because I didn't want to be too aggressive or discourage people from doing open-source development... but the data speaks for itself).

#### The stats were as of 2024-09-13 (now outdated)
- **Only about 0.027% of downloads had resulted in a donation so far.**  
    (considering about 26140 downloads in both VS Marketplace and Open-VSX since the donation link was added)
    I'm aware not every download is a unique user, but even multiplying by 10 (number of versions) it's still just 0.27%. For reference Thunderbird claims to receive donations from 3% of their users (yes, it's a absurdly much bigger cooler project, but my point is mine is just not worth a lot, and thus the rational thing is to abandon it).
- The "received" amount (7x$3=$21 donations at the time, minus fees/taxes of about 30~40% per transaction, and itch's 10% cut I had set at the time) means **only about US$0.18 per commit** (1 commit is roughly an entire day of work, sometimes more, sometimes less).  
    (considering 77 commits published since the donation link was added; value is after deducing taxes and fees)
- Well, that ***would*** be the received amount, except ***it was actually zero***. Blame Payoneer's account maintenance fees (US$29.95/year) and minimum withdrawal limit (US$150). So the remaining part of the donations were so little that they were actually entirely consumed by bank fees. Yay.

TL;DR If you make a rational analysis, my conclusion would have to be: DO NOT work on open-source unless:
- you live in the US and can easily receive dollars with less fees; or
- you have enough of a passive income to afford it; or
- you have a regular job, and do it only on your free time; or
- you're already rich

This isn't my case, and I've dedicated plenty of time at this -- please note how all reviews are 5 stars and nobody has ever complained about a bug so far.

### Is this crowdfunding?
> Crowdfunding isn't about threatening to abandon the project!

I suppose you're right. Perhaps I shouldn't call this crowdfunding, as that term is usually more for when you DON'T have a product yet published, and you just want money for a ***promise*** that you'll develop it, to be given to you before you have anything ready to reward your patrons.

This isn't what I'm doing. I just want to be compensated by my previous work enough that I can justify this project's existence. And by making a one-time small payment (which I should hopefully receive a fraction of) you can access restricted features more conveniently, instead of having to wait or build the extension yourself.

**Comment by LokyStaz** [comment]
That is a full detailed answer ^^ thanks! I really appreciate that you take the time to answer.
I understand better your situation and why the "paywall".
And I am truly sorry because I'm in the same position as you.
I mean a solo dev without social skill nor money, who try to survive in this harsh world.
And I'm even more sorry because I don't have time right now to help you on this project!

But after reading this, I think that instead of speaking about "abandon" in the README,
it would be interesting to be more specific, something like:

> "Development of new features will be interrupted and will pick up again later,
> once I’ve wrapped up my other, more important projects" !

I wish you all the best,
and do not hesitate to contact me if you need some help in your dev journey!

**Comment by AlfishSoftware** [comment]
Thanks, I appreciate the understanding, and how you can relate. And just in case, I apologize if anything might have sounded rude, as it isn't my intention. I totally respect and appreciate how cool free software projects are, and I would love to let everyone have easy access to plenty of features if I could afford to.

And to clarify something I may have phrased badly, I'm not trying to blame others for not receiving any help in this project. Although I wouldn't reject receiving help, it's also true that I've never explicitly wrote something like "contributions welcome" because:

1. I'm (to my dismay) too much of a perfectionist, so it's really difficult to trust other people's code would be up to my own quality standards I have for myself.
2. And honestly, financial donations would help me much more than donations in code or ideas. I already had in my head a clear vision of the ways I can improve this extension, and features that I know I would want to and be able to implement (as mentioned at the very bottom in the README, plus plenty of other bigger ideas I never wrote there) if I had enough time that I could put to work on this.

Sadly I'm not in a position where I can afford to get distracted from my project, or maintain a near-zero revenue overall any longer, so I had to reflect in this project the ultimatum I've received... mostly for my own focus.

Even if I end up abandoning this, if I ever get in a place where I can get significant passive income, I would be glad to resurrect it to implement those features; after all, I'm doing this mostly for my own use.

Thanks a lot for the feedback, and I wish you all success as well! 😊 :rocket:
